### ISSUE TITLE
Not detected "patch < %{PATCHX}"

### ISSUE THREAD

**Issue by junaruga** [rpm-software-management/rpmlint#91]
Right now `rpmlint` detects below patch logic in the check section.

```
cat %{PATCHX} | patch
```

However below logic is not detected.

```
patch < %{PATCHX}
```

Could you implement for `rpmlint` to detect this case too?

I used below spec file to check this.
http://pkgs.fedoraproject.org/cgit/rpms/rubygem-activesupport.git/tree/rubygem-activesupport.spec

```
%check
...
cat %{PATCH1} | patch -p2
cat %{PATCH2} | patch -p2
patch -p2 < %{PATCH4}
cat %{PATCH7} | patch -p2
cat %{PATCH9} | patch -p2
cat %{PATCH10} | patch -p2
```

```
$ DESTDIR="./dist" make install

$ dist/usr/bin/rpmlint ~/fed/rubygem-activesupport/rubygem-activesupport.spec
/home/jaruga/fed/rubygem-activesupport/rubygem-activesupport.spec: W: patch-not-applied Patch4: rubygem-activesupport-5.0.1-Fixnum-and-Bignum-are-deprecated-in-Ruby-trunk-tests.patch
/home/jaruga/fed/rubygem-activesupport/rubygem-activesupport.spec: W: invalid-url Source1: activesupport-5.0.1-tests.tgz
0 packages and 1 specfiles checked; 0 errors, 2 warnings.
```

Thanks.

**Comment by junaruga** [comment]
Thanks!

**Comment by junaruga** [comment]
@scop 
Sorry still rpmlint can not detect it.

```
$ which rpmlint
/home/jaruga/git/rpmlint/dist/usr/bin/rpmlint

$ rpmlint rubygem-activesupport.spec 
rubygem-activesupport.spec: W: patch-not-applied Patch4: rubygem-activesupport-5.0.1-Fixnum-and-Bignum-are-deprecated-in-Ruby-trunk-tests.patch
rubygem-activesupport.spec: W: invalid-url Source1: activesupport-5.0.1-tests.tgz
0 packages and 1 specfiles checked; 0 errors, 2 warnings.
```

**Comment by scop** [comment]
Current git version works just fine for me for your test specfile. If it didn't, it would also fail the CI tests we run at Travis. Try invoking `./rpmlint -C . ~/fed/rubygem-activesupport/rubygem-activesupport.spec` in your rpmlint git clone dir.

**Comment by junaruga** [comment]
@scop 

Oh it works in my rpmlint git clone dir! Umm, why..

```
$ ./rpmlint -C . ~/fed/rubygem-activesupport/rubygem-activesupport.spec
/home/jaruga/fed/rubygem-activesupport/rubygem-activesupport.spec: W: invalid-url Source1: activesupport-5.0.1-tests.tgz
0 packages and 1 specfiles checked; 0 errors, 1 warnings.
```

**Comment by junaruga** [comment]
Below works.

```
$ ./dist/usr/bin/rpmlint -C . ~/fed/rubygem-activesupport/rubygem-activesupport.spec/home/jaruga/fed/rubygem-activesupport/rubygem-activesupport.spec: W: invalid-url Source1: activesupport-5.0.1-tests.tgz
0 packages and 1 specfiles checked; 0 errors, 1 warnings.
```

Below does not work.

```
$ ./dist/usr/bin/rpmlint ~/fed/rubygem-activesupport/rubygem-activesupport.spec
/home/jaruga/fed/rubygem-activesupport/rubygem-activesupport.spec: W: patch-not-applied Patch4: rubygem-activesupport-5.0.1-Fixnum-and-Bignum-are-deprecated-in-Ruby-trunk-tests.patch
/home/jaruga/fed/rubygem-activesupport/rubygem-activesupport.spec: W: invalid-url Source1: activesupport-5.0.1-tests.tgz
0 packages and 1 specfiles checked; 0 errors, 2 warnings.
```

**Comment by scop** [comment]
That's expected. The "not work" commands are not running the rpmlint checker modules from your rpmlint git checkout, but probably from `/usr/share/rpmlint`.

**Comment by junaruga** [comment]
My python environment as a reference.

```
$ which python
/usr/local/python-2.7.13/bin/python

$ which python3
/usr/local/python-3.6.0/bin/python3
```

**Comment by junaruga** [comment]
> That's expected. The "not work" commands are not running the rpmlint checker modules from your rpmlint git checkout, but probably from /usr/share/rpmlint.

Oh okay I did not know that.
Thank you.
